### ISSUE TITLE
Trashed files & bookmarks & other applications like Working Copy

### ISSUE THREAD

**Issue by zrzka** [omz/Pythonista-Issues#519]
**System Information**

* Pythonista 3.2 (320000), Default interpreter 3.6.1
* iOS 11.2.2, model iPad7,2, resolution (portrait) 2048.0 x 2732.0 @ 2.0

**Affects**

Pythonista >= 3.2 & iOS 11 only.

**Description**

Prior iOS 11, there was no issue with trashing files. But since we have features like Open in place... now, trashed file behavior should be rethinked? It's more like usability issue. Try this:

* Open folder (repository) from Working Copy
* Trash some file from this folder (`new_tab.py` in my case)
* Open Working Copy
* `new_tab.py` is gone, but there's new file named `.new_tab.py.icloud`, which is treated like any other regular file by Working Copy - added

To fix this, I have to trash file in Pythonista and then empty trash to get rid of this file. Basically it forces more to do more than just simple swipe & trash.

![91b7fb76-1050-4cef-84e7-8f6befae7b39](https://user-images.githubusercontent.com/1084172/35093435-aa0b46ae-fc42-11e7-99b8-9c7adbc3fc93.png)

**Expected behavior**

I would expect that when I trash file in Pythonista the file disappears from same folder and no new file is introduced in the same location. Just to avoid clashes with other applications.

iOS 11 introduced `NSTrashDirectory`, tried it, but it returns nothing (simulator, didn't try it on device). Probably `NSCachesDirectory` can be used? But this directory isn't part of the backup process. So, if backup is required for trashed files, `NSDocumentDirectory` should be used.

Maybe `.Trash-iCloud`, `.Trash-bookmark` can be introduced? When I trash file:

* `This iPad` -> `.Trash`
* `iCloud` -> `.Trash-iCloud`
* `EXTERNAL FILES` -> `.Trash-bookmark`.

Just thinking aloud, marked as minor, but still thinks that this should be addressed somehow.

**Comment by zrzka** [comment]
Removed `enhancement` & `minor` and added `bug`. Looks like more serious issue. Try to do this:

* Open Working Copy
* Clone any repository
* Open Pythonista
* _EXTERNAL FILES_ - _Open..._ (`blackmamba` for example)
* Add new file `blackmamba/test.txt`
* Write some nonsense & close this file
* Open this file again via long press on Pythonista dock icon
* Close the file
* Delete this file in Pythonista
* Empty trash in Pythonista
* Kill Pythonista
* Open Working Copy
* There's no file `blackmamba/.test.txt.icloud` - ok
* Remove repository from Working Copy
* Clone it again
* Launch Pythonista, just launch it
* Open Working Copy
* And the file `blackmamba/.test.txt.icloud` is back

Basically when I trash file from _EXTERNAL FILES_ folder, the file is reappearing no matter what I do. It reappears only & only when I kill Pythonista & launch fresh instance. When the Pythonista is not killed, everything's okay.

![ff011bb5-6502-42c9-a4e2-a4af88a838d3](https://user-images.githubusercontent.com/1084172/35152300-2f16aa5c-fd22-11e7-80b4-2f65992d71ab.png)

**Comment by zrzka** [comment]
As a workaround, I have to use [.gitignore](https://github.com/zrzka/blackmamba/commit/c37e3bebe1bfc285ed0c6067ac737cf4bf41688c).
